### PR TITLE
feat(ui): add ability to create RAID 0 storage devices

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added ability to create RAID 0 devices. The rest of the RAID levels and the storage devices table will come in a later PR.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a Ready or Allocated machine
- Create two unformatted partitions
- Select them both and check that the "Create RAID" button becomes enabled
- Click "Create RAID" and change some of the values
- Check that a RAID 0 device is created with the values you entered

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2166

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/BulkActions/CreateRaid/CreateRaid.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                  diskIds.push(storageDevice.id);
+                } else {
+                  partitionIds.push(storageDevice.id);
+                }
+                return [diskIds, partitionIds];
+              },
+              [[], []]
+            );
+            const params = {
+              // TODO: Replace with dynamic value
```
Comment (huwshimi) [inline ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/BulkActions/CreateRaid/CreateRaid.tsx:104]: Is this a todo for a future branch?

File: ui/src/app/store/machine/utils/storage.ts [hunk 2]
```diff
@@ -284,7 +314,7 @@ export const isMounted = (fs: Filesystem | null): fs is Filesystem => {
   // VMware ESXi does not directly mount the partitions used. As MAAS can't
   // model that, a placeholder "RESERVED" is used for datastores so we know that
   // these partitions are in use.
-  return fs.mount_point !== "" && fs.mount_point !== "RESERVED";
+  return ![null, "", "RESERVED"].includes(fs.mount_point);
```
Comment (huwshimi) [inline ui/src/app/store/machine/utils/storage.ts:317]: Could `mount_point` also be undefined? Could this be `!!fs.mount_point && fs.mount_point !== "RESERVED";`?

**Review by huwshimi — APPROVED** [review]
A couple of questions, but looks good!

File: ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/BulkActions/CreateRaid/CreateRaid.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/BulkActions/CreateRaid/CreateRaid.tsx:104]: Yep - I decided to break down the Create RAID issue into two parts so I'll start working on that pretty much straight away.

File: ui/src/app/store/machine/utils/storage.ts (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/machine/utils/storage.ts:317]: Yeah I think a catch-all for falsy values is probably better!

**Pull request merged**